controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -556,6 +556,15 @@ bool wfa_ca::create_cmdu(ieee1905_1::CmduMessageTx &cmdu_tx, ieee1905_1::eMessag
     int tlv_type;
 
     switch (message_type) {
+
+    case ieee1905_1::eMessageType::AP_METRICS_QUERY_MESSAGE: {
+        cmdu_header = cmdu_tx.create(g_mid, ieee1905_1::eMessageType::AP_METRICS_QUERY_MESSAGE);
+        if (!cmdu_header) {
+            LOG(ERROR) << "Failed building message!";
+            return false;
+        }
+        break;
```

Comment:
In the standard, it is mentioned that AP METRICS QUERY must have one AP METRICS QUERY TLV (0x93), however, it does not exist on the test sequence.
I'll guess you need to understand how to add and fill this TLV, relying on data you already have.

Reply:
Sure but this  is confusing , 
I think that all messages with or without TLVs can be handled automatically and  create_cmdu()  needs to handle only CHANNEL_SELECTION_REQUEST_MESSAGE

Reply:
Here I fixed it for you: #258